utils/hare-status

```diff
@@ -167,31 +148,91 @@ def processes(cns: Consul, consul_util: ConsulUtil,
         parts = x['Key'].split('/', 5)
         if len(parts) == 5:
             proc_fidks.append(fid_key(parts[-1]))
+    node_health = consul_util.get_node_health_status(node_name)
+    process_states = get_node_process_states(cns, node_name)
+    process_list = get_node_processes(cns, node_name)
     return [
         Process(
-            name=proc_id2name(cns, node_name, k),
+            name=proc_id2name(process_list, k),
             fid=Fid(0x7200000000000001, k),
-            ep=(
-                kv_value_as_str(
-                    cns,
-                    f'm0conf/nodes/{node_name}/processes/{k}/endpoint'
-                ) or '**ERROR**'),
-            status=process_status(consul_util, node_name,
+            ep=(proc_id2endpoint(process_list, k) or '**ERROR**'),
+            status=process_status(node_health, process_states,
                                   Fid(0x7200000000000001, k)))
         for k in sorted(proc_fidks)
     ]
 
 
-def process_status(consul_util: ConsulUtil, node_name, fid: Fid) -> str:
-    if (consul_util.get_node_health_status(node_name) != 'passing'):
+def process_status(node_health: str, proc_states: Dict[str, str],
+                   fid: Fid) -> str:
+    if node_health != 'passing':
         return 'offline'
-    status = consul_util.get_process_status(fid, node_name)
-    if status.proc_status == 'M0_CONF_HA_PROCESS_STARTED':
-        return 'started'
-    elif status.proc_status in ('Unknown', 'UNKNOWN'):
-        return 'unknown'
+    if str(fid) in proc_states:
+        return proc_states[str(fid)]
     else:
-        return 'offline'
+        return 'unknown'
+
+
+def get_node_process_states(cns: Consul, node_name: str) -> Dict[str, str]:
+    key = f'{node_name}/processes/'
```

Comment:
this **key = f'{node_name}/processes/'** fetches below format key
```
cortx-data-headless-svc-ssc-vm-g3-rhev4-2387/processes/0x7200000000000001:0x27:{"state": "M0_CONF_HA_PROCESS_STARTED", "type": "M0_CONF_HA_PROCESS_M0D"}
cortx-data-headless-svc-ssc-vm-g3-rhev4-2387/processes/0x7200000000000001:0x2a:{"state": "M0_CONF_HA_PROCESS_STARTED", "type": "M0_CONF_HA_PROCESS_M0D"}
cortx-data-headless-svc-ssc-vm-g3-rhev4-2387/processes/0x7200000000000001:0x31:{"state": "M0_CONF_HA_PROCESS_STARTED", "type": "M0_CONF_HA_PROCESS_M0D"}
```

and `get_node_processes()` fetches
```
m0conf/nodes/cortx-data-headless-svc-ssc-vm-g3-rhev4-2387/processes/52/meta_data:/dev/sdc
m0conf/nodes/cortx-data-headless-svc-ssc-vm-g3-rhev4-2387/processes/52/services/addb2:61
m0conf/nodes/cortx-data-headless-svc-ssc-vm-g3-rhev4-2387/processes/52/services/cas:62
m0conf/nodes/cortx-data-headless-svc-ssc-vm-g3-rhev4-2387/processes/52/services/fdmi:64
```
Both the keys are different. So we cannot use **get_node_processes()** here.
Even the prefix isn't the same for both keys... So is there any other way to use it?

Reply:
Okay, It would be good to avoid pattern matching as the cluster and the number keys grow, the data for pattern matching will increase, pattern matching can be expensive as well. If we have the information in a hashed format (associative array) then it becomes easier.